Automattic/wp-calypso
# Editor Deprecation: Consolidate the editor URL selectors v2

#### Changes proposed in this Pull Request

This is an updated version of #47344 which needed to be reverted as it
didn't take into account the patch that had been added to prevent us
from Calypsoifying Jetpack 9.2 sites #47925.

As the Jetpack version test needs to happen in two places, I've
abstracted it to a new selector.

Having this in place should simplify the process of disabling
Calypsoify.

#### Testing instructions

* Check that links to edit posts and pages don't change for simple sites, and sites that are eligible for Gutenframe, with this change.
* For Jetpack (including Atomic) sites that are ineligible for Gutenframe, check that the calypsoify=1 parameter is added to the editor URLs.
* Check that Jetpack sites with version 9.2 use standard /wp-admin/post.php and post-new.php URLs to load the editor, with no `calypsoify` parameter. Atomic sites were unaffected by the problem, but they will no longer be running 9.2
